src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java

```diff
@@ -591,4 +645,50 @@ private String getVariableValue(Iterator<Object> iterator, String variableName)
         logger.info("Variable {} found, value: {}", variableName, value);
         return value;
     }
-}
+
+    private TransactionStatusResponseDTO setTxnFound(TransactionStatusResponseDTO response, JSONObject transfer) {
+        String status = transfer.has("status")  ? transfer.getString("status") :
+                transfer.getString("state");
+        response.setCompletedTimestamp(transfer.isNull("completedAt") ? null :
```

Comment:
There are many instances of this type of null check in the code, please check if we can use the NVL method, or use a utility method to do this as this is a low-level comparison and it is repeating. this needs to be a util function.